plugin/ale.vim

```diff
@@ -2,41 +2,66 @@
 " Description: Main entry point for the plugin: sets up prefs and autocommands
 "   Preferences can be set in vimrc files and so on to configure ale
 
+" Sanity Checks
+
 if exists('g:loaded_ale')
     finish
 endif
 let g:loaded_ale = 1
 
 " A flag for detecting if the required features are set.
 if has('nvim')
-    let g:ale_has_required_features = has('timers')
+    let s:ale_has_required_features = has('timers')
 else
-    let g:ale_has_required_features = has('timers') && has('job') && has('channel')
+    let s:ale_has_required_features = has('timers') && has('job') && has('channel')
 endif
 
-if !g:ale_has_required_features
+if !s:ale_has_required_features
     echoerr 'ALE requires NeoVim >= 0.1.5 or Vim 8 with +timers +job +channel'
     echoerr 'Please update your editor appropriately.'
     finish
 endif
 
+" Globals
+
+let g:ale_buffer_loclist_map = {}
+let g:ale_buffer_should_reset_map = {}
+let g:ale_buffer_sign_dummy_map = {}
+
+" User Configuration
+
 " This list configures which linters are enabled for which languages.
 let g:ale_linters = get(g:, 'ale_linters', {})
 
-" This flag can be set to 0 to disable linting when text is changed.
-let g:ale_lint_on_text_changed = get(g:, 'ale_lint_on_text_changed', 1)
-
 " This flag can be set with a number of milliseconds for delaying the
 " execution of a linter when text is changed. The timeout will be set and
 " cleared each time text is changed, so repeated edits won't trigger the
 " jobs for linting until enough time has passed after editing is done.
 let g:ale_lint_delay = get(g:, 'ale_lint_delay', 200)
 
+" This flag can be set to 0 to disable linting when text is changed.
+if get(g:, 'ale_lint_on_text_changed', 1)
```

Comment:
These changes won't set the global variables to a default value. That can be a problem if you want to read them in other files. I'd keep setting these flags to their default value.